tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ManagedExecutorTest.java

```diff
@@ -427,6 +432,435 @@ public void contextControlsForManagedExecutorBuilder() throws InterruptedExcepti
+                Assert.assertEquals(Label.get(), "contextualFunctionOverride-label-1",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type not cleared from thread.");
+                return i + 1;
+            });
+
+            Buffer.set(new StringBuffer("contextualFunctionOverride-buffer-2"));
+            Label.set("contextualFunctionOverride-label-2");
+
+            Function<Integer, Integer> precontextualizedFunction2 = labelContext.contextualFunction(i -> {
+                Assert.assertEquals(Label.get(), "contextualFunctionOverride-label-2",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type not cleared from thread.");
+                return i + 20;
+            });
+
+            Function<Throwable, Integer> precontextualizedErrorHandler = labelContext.contextualFunction(failure -> {
+                Assert.assertEquals(Label.get(), "contextualFunctionOverride-label-2",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type not cleared from thread.");
+                return -1;
+            });
+
+            Buffer.set(new StringBuffer("contextualFunctionOverride-buffer-3"));
+            Label.set("contextualFunctionOverride-label-3");
+
+            Function<Integer, Integer> normalFunction = i -> {
+                Assert.assertEquals(Buffer.get().toString(), "contextualFunctionOverride-buffer-3",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Label.get(), "",
+                        "Context type not cleared from thread.");
+                return i + 300;
+            };
+
+            CompletableFuture<Integer> stage0 = executor.newIncompleteFuture();
+            CompletableFuture<Integer> stage1 = stage0.thenApplyAsync(precontextualizedFunction1);
+
+            Buffer.set(new StringBuffer("contextualFunctionOverride-buffer-4"));
+            Label.set("contextualFunctionOverride-label-4");
+
+            Function<Integer, CompletableFuture<Integer>> precontextualizedFunction4 = labelContext.contextualFunction(i -> {
+                Assert.assertEquals(Label.get(), "contextualFunctionOverride-label-4",
+                        "Previously captured context type not found on thread.");
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type not cleared from thread.");
+                return stage1;
+            });
+
+            Buffer.set(new StringBuffer("contextualFunctionOverride-buffer-3"));
+            Label.set("contextualFunctionOverride-label-3");
+
```

Comment:
Did you intend to reuse these context strings? "contextualFunctionOverride-*-3"

Reply:
yes, otherwise the test logic wouldn't be correct.  See lines 683/657.